build/common.sh

```diff
@@ -596,70 +653,134 @@ function kube::build::run_build_command() {
 
   # Clean up container from any previous run
   kube::build::destroy_container "${KUBE_BUILD_CONTAINER_NAME}"
-  "${docker_cmd[@]}" "$@"
+  "${docker_cmd[@]}" "${cmd[@]}"
   kube::build::destroy_container "${KUBE_BUILD_CONTAINER_NAME}"
 }
 
-# Test if the output directory is remote (and can only be accessed through
-# docker) or if it is "local" and we can access the output without going through
-# docker.
-function kube::build::is_output_remote() {
-  rm -f "${LOCAL_OUTPUT_SUBPATH}/test_for_remote"
-  kube::build::run_build_command touch "${REMOTE_OUTPUT_BINPATH}/test_for_remote"
+function kube::build::probe_address {
+  # Apple has an ancient version of netcat with custom timeout flags.  This is
+  # the best way I (jbeda) could find to test for that.
```

Comment:
Totally agree that we should minimize this.  The stuff in `build/` is needed to enable us to easily get there.

My guide: `build/*` should work without having to install anything special besides Docker for Mac.  That is super easy and we can almost assume it at this poing.

Stuff in `hack/*` should feel free to make linux-y/GNU assumptions.  If users *really* want to run on OS X it'll be unsupported and they'll probably have to `brew install` some stuff.  Most users will be told to just do a `build/run.sh <command>` instead.